### PR TITLE
Add glossary of projects and edit landing page docs

### PR DESCRIPTION
**Pull request by willingc**
Addresses #25.

This PR adds an alphabetical glossary of all Jupyter related projects to assist a new user to the Jupyter ecosystem.

There are edits to the landing page as well. I'm happy to change anything and incorporate suggestions.

### CODE REVIEW

**Comment by Carreau** [comment]
+10.

Want to ping @ellisonbg to give him  chance to have a look before merging.

**Comment by ellisonbg** [comment]
Thanks I will have a look!

File: docs/source/index.rst [hunk 1]
```diff
@@ -20,71 +22,100 @@ and IPython organizations.
...
+  for organizations with plugglable authentication and scalability.
+* `jupyter-drive <https://github.com/jupyter/jupyter-drive>`__: store notebooks
+  on Google Drive.
+* `nbviewer <https://github.com/jupyter/nbviewer>`__: share notebooks as static
+  HTML on the web.
+* `tmpnb <https://github.com/jupyter/tmpnb>`__: create temporary, transient
+  notebooks in the cloud.
+* `dockerspawner <https://github.com/jupyter/dockerspawner>`__: deploy
+  notebooks for `jupyterhub` inside Docker containers.
+* `tmpnb-deploy <https://github.com/jupyter/tmpnb-deploy>`__: deployment tools
```
Comment (rgbkrk) [inline docs/source/index.rst]: The selfish piece of tmpnb-deploy is that's it's more geared towards our own deployment and not composable pieces for tmpnb deployments in general. One can adapt things from them, as we've witnessed from people, but standalone they're pretty heavy on opinion as we've figured them out.

I'm not sure if this means that I'm saying we shouldn't include it on the list or if it should say "deployment tools for the tmpnb backing try.jupyter.org"

**Comment by rgbkrk** [comment]
+:100: Thank you so much for this!

File: docs/source/index.rst [hunk 2]
```diff
@@ -20,71 +22,100 @@ and IPython organizations.
...
-- `Jupyter Notebook <http://jupyter-notebook.readthedocs.org/en/latest/>`__: web-based application for authoring documents that combine live-code with narrative text, equations and visualizations.
-- `Jupyter Console <https://github.com/jupyter/jupyter_console>`__: terminal based console for interactive computing.
-- `Jupyter QtConsole <https://github.com/jupyter/qtconsole>`__: Qt application for interactive computing with rich output.
 
 Kernels
 -------
 
-Kernels are separate processes that sit behind the Jupyter user interfaces and
-run code in a particular programming language.
+Kernels are `programming language specfic` processes that run independently
```
Comment (Carreau) [inline docs/source/index.rst]: specfic -> specific.

File: docs/source/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    `ipyparallel <http://ipyparallel.readthedocs.org/en/latest/>`__
+        lightweight parallel computing in Python offering seamless notebook integration.
+
+    `IPython <http://ipython.org>`__
+        the reference Jupyter kernel, providing a powerful environment for
+        interactive computing in Python.
+
+    `IPython Kernel <http://ipython.readthedocs.org/en/master/>`__
+        interactive computing in Python.
```
Comment (Carreau) [inline docs/source/glossary.rst]: Flipped description for IPython and kernel.

**Comment by Carreau** [comment]
Merging, I'll send a PR to fix the comments I made.

**Comment by willingc** [comment]
@Carreau Thanks a bunch.

@Carreau, @rgbkrk, @ellisonbg - I have some travel coming up so if there are any urgent docs on the wishlist do let me know.

**Comment by Carreau** [comment]
> @Carreau Thanks a bunch.

Thanks to you !

> @Carreau, @rgbkrk, @ellisonbg - I have some travel coming up so if there are any urgent docs on the wishlist do let me know.

Any docs update is appreciated :-) I try to think and ping you on something specific is needed !

**Pull request merged**